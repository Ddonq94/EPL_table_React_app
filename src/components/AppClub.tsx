import { getTeamData } from "../services/dataService";

function AppClub({ testID = "AppClub", clubName, justifyEnd = false }: any) {
  return (
    <div
      data-testid={testID}
      className={`club-wrapper ${justifyEnd ? "justify-end" : ""} `}
    >
      <img src={getTeamData(clubName).logo} alt={`${clubName}logo`} />
      <span> {clubName}</span>
    </div>
  );
}

export default AppClub;
