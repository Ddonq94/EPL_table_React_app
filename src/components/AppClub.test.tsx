import React from "react";
import { cleanup, render } from "@testing-library/react";
import AppClub from "./AppClub";

afterEach(() => {
  cleanup();
});

test("renders component", () => {
  const TestFn = () => {
    return <AppClub testID="AppClub" />;
  };

  const { getByTestId } = render(<TestFn />);
  const content = getByTestId("AppClub");
  expect(content).toBeDefined();
  expect(content).toBeInTheDocument();
  expect(content).toBeDefined();
  expect(content).toBeEnabled();
  expect(content).not.toBeEmptyDOMElement();
});
