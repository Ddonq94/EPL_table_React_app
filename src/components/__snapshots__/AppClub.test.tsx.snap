// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Club rendering renders AppClub to match snapshot 1`] = `
<div
  class="club-wrapper  "
  data-testid="AppClub"
>
  <img
    alt="LiverpoolLogo"
    data-testid="AppClubImg"
    src="https://resources.premierleague.com/premierleague/badges/25/t14.png"
  />
  <span>
     
    Liverpool
  </span>
</div>
`;
